File: SCG Addon Taller/MuestraMessgeBoxSBO.vb

Imports SAPbouiCOM

Public Class MuestraMessgeBoxSBO
    Private _sboApplication As SAPbouiCOM.Application

    Public Sub New(ByVal sboApplication As Application)
        _sboApplication = sboApplication
    End Sub

    Public Function MessageBxPreg(ByVal mensaje As String) As Boolean
        Return _sboApplication.MessageBox(Text:=mensaje, Btn1Caption:=My.Resources.Resource.Si, Btn2Caption:="No", Btn3Caption:="") = 1
    End Function

    Public Sub MessageBxExc(ByVal mensaje As String)
        _sboApplication.MessageBox(Text:=mensaje, Btn1Caption:="Ok")
    End Sub

End Class

File: SCG.ServicioPostVenta/MuestraMessgeBoxSBO.cs
using System;
using SAPbouiCOM;

namespace SCG.ServicioPostVenta
{
    public class MuestraMessgeBoxSBO
    {
        private static SAPbouiCOM.Application _sboApplication;

        public MuestraMessgeBoxSBO(SAPbouiCOM.Application sboApplication)
        {
            _sboApplication = sboApplication;
            //Del g = MessageBxPreg;
        }

        public delegate void Del(string message);

        public bool MessageBxPreg(String mensaje)
        {
            return _sboApplication.MessageBox(Text: mensaje, DefaultBtn: 1, Btn1Caption: Resource.Si, Btn2Caption: "No", Btn3Caption: "") == 1;
        }

        public void MessageBxExc(string mensaje)
        {
            _sboApplication.MessageBox(Text: mensaje, DefaultBtn: 1, Btn1Caption: "Ok");
        }
    }
}
